test/test/test_conn_primitive_serial.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2018-2021, Arm Limited and affiliates.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+import pytest
+
+from mbed_os_tools.test.host_tests_conn_proxy.conn_primitive_serial import SerialConnectorPrimitive
+
+@mock.patch("mbed_os_tools.detect.create")
+def test_provided_serial_port_used_with_target_id(mock_create):
```

Comment:
You will need to create a test class inheriting from `unittest.TestCase` and add 

```
if __name__ == '__main__':
      unittest.main()
```

because CI doesn't use the `pytest` runner.